internal/app/backend/backend_service_test.go

```diff
@@ -0,0 +1,141 @@
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{Assignment: &pb.Assignment{}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{TicketId: []string{xid.New().String(), xid.New().String()}}, func(_ *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{xid.New().String(), xid.New().String()},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(codes.NotFound, status.Convert(err).Code())
+	})
+}
+
+func TestAssignTicketsNormalRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	fe := pb.NewFrontendClient(tc.MustGRPC())
+
+	ctResp, err := fe.CreateTicket(tc.Context(), &pb.CreateTicketRequest{Ticket: &pb.Ticket{}})
+	assert.Nil(err)
+
+	req := &pb.AssignTicketsRequest{
+		TicketId: []string{ctResp.Ticket.Id},
+		Assignment: &pb.Assignment{
+			Connection: "localhost",
+		},
+	}
+	assignTicketsLoop(tc, req, func(resp *pb.AssignTicketsResponse, err error) {
+		assert.Equal(&pb.AssignTicketsResponse{}, resp)
+	})
+
+	gtResp, err := fe.GetTicket(tc.Context(), &pb.GetTicketRequest{TicketId: ctResp.Ticket.Id})
+	assert.Nil(err)
+	assert.Equal("localhost", gtResp.Assignment.Connection)
+}
+
+func TestFetchMatchesEmptyRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	fetchMatchesLoop(t, tc, &pb.FetchMatchesRequest{}, func(_ *pb.FetchMatchesResponse, err error) {
+		assert.Equal(codes.InvalidArgument, status.Convert(err).Code())
+	})
+}
+
+// TODO: Add FetchMatchesNormalTest when Hostname getter used to initialize mmf service is in
+// https://github.com/GoogleCloudPlatform/open-match/pull/473
+func fetchMatchesLoop(t *testing.T, tc *rpcTesting.TestContext, req *pb.FetchMatchesRequest, handleResponse func(*pb.FetchMatchesResponse, error)) {
+	c := pb.NewBackendClient(tc.MustGRPC())
+	stream, err := c.FetchMatches(tc.Context(), req)
+	if err != nil {
+		t.Fatalf("error querying tickets, %v", err)
+	}
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+		handleResponse(resp, err)
+		if err != nil {
+			return
+		}
+	}
+}
+
+func assignTicketsLoop(tc *rpcTesting.TestContext, req *pb.AssignTicketsRequest, handleResponse func(*pb.AssignTicketsResponse, error)) {
+	c := pb.NewBackendClient(tc.MustGRPC())
+	resp, err := c.AssignTickets(tc.Context(), req)
+	handleResponse(resp, err)
+	if err != nil {
+		return
+	}
+}
+
```

Comment:
This test coverage is great - Lets discuss offline where it must live though. If this test relies on frontend and mmlogic to be up, then should it be a part of E2E?